### ISSUE TITLE
LocalReadoutMitigator does not support BackendV2

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit#12832]
### Environment

- **Qiskit version**: 1.3.0.dev0+6b521db
- **Python version**: 3.9
- **Operating system**: MacOS

### What is happening?

The class `qiskit.result.mitigation.local_readout_mitigator.LocalReadoutMitigator` does not support BackendV2. Given that [BackendV1 is now deprecated](https://github.com/Qiskit/qiskit/pull/12629)... it getting complicated to have this class post 2.0. 

### How can we reproduce the issue?

```python
from qiskit.result import LocalReadoutMitigator
from qiskit.providers.fake_provider import GenericBackendV2

backend = GenericBackendV2(
    num_qubits=5,
    seed=42,
)
LocalReadoutMitigator(backend=backend)
```
```
AttributeError: 'GenericBackendV2' object has no attribute 'properties'
```

### What should happen?

The class works with BackendV1
```python
from qiskit.result import LocalReadoutMitigator
from qiskit.providers.fake_provider import Fake5QV1

LocalReadoutMitigator(backend=Fake5QV1())
```
Ideally, it should also work with BackendV2.

### Any suggestions?

`¯\_(ツ)_/¯`

**Comment by 1ucian0** [comment]
Reducing priority to `medium` as `qiskit.result.mitigation` is going to be removed in 2.0 https://github.com/Qiskit/qiskit/issues/12838 . Still would be great to have `BackendV2` support in `LocalReadoutMitigator`, as it is supported for the full 1.* cycle.

**Comment by ElePT** [comment]
This issue will be closed with #13351 (deprecating the class).
